### PR TITLE
Bi-directional limits for functions with test

### PR DESCRIPTION
**Pull request by apalmer1377**
Added bi-directional functionality to limits, set the default to the bi-directional option, and added some tests.  This is my first pull request to a public project, so sorry if there's something wrong with it!

### CODE REVIEW

File: sympy/series/limits.py [hunk 1]
```diff
@@ -42,7 +42,15 @@ def limit(e, z, z0, dir="+"):
...
     """
 
-    return Limit(e, z, z0, dir).doit(deep=False)
+    if dir == "+-":
+        llim = Limit(e, z, z0,dir="-").doit(deep=False)
+        rlim = Limit(e, z, z0,dir="+").doit(deep=False)
+        if llim == rlim:
+            return rlim
+        else:
+            return None
```
Comment (asmeurer) [inline sympy/series/limits.py]: I don't think None is an appropriate return type. It should either raise an exception or return a SymPy type.

**Comment by apalmer1377** [comment]
Changed the code so that the limit raises a ValueError when the right- and left-handed limits don't equal each other.

**Comment by asmeurer** [comment]
I like the change. There has been some debate here, though, so I want to leave this for a bit to see if anyone else has any comments.

**Review by asmeurer — APPROVED** [review]

**Comment by latot** [comment]
Hi, well you know what i want say, little question, why you don't like ```None```?, is used in some functions like iscomplement and similars.
(Raise exception break the workflow of the program)

**Comment by apalmer1377** [comment]
I think both approaches make sense--raising an exception is probably more useful in practice, but `None` seems more in line with the mathematical definition.  There could be an argument in the `limit` function that defaults to raising an exception, but can be changed to return `None` instead, like:

```
if except == True:  //except is the name of the argument
    raise ValueError
else:
    return None
```

**Comment by latot** [comment]
seems a nice idea!

**Comment by latot** [comment]
Little question, what is the focus of Sympy in the sense of math works?, this is because raise an exception it means Sympy will works be default with continuous functions, personally one of the things i like is you can works with non continuous functions here.

**Comment by moorepants** [comment]
This is passing tests. Is it ready to merge?

**Comment by gxyd** [comment]
Closing this. See #13463 , which took the necessary commits from this PR.

**Pull request closed without merging**